auto_remediate/test/test_securityhub_s3.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+import logging
+
+import moto
+import pytest
+
+from .. import security_hub_rules
+
+
+class TestSecurityHubS3BucketPublicReadProhibited:
+    @pytest.fixture
+    def sh(self):
+        with moto.mock_s3():
+            sh = security_hub_rules.SecurityHubRules(logging)
+            yield sh
+
+    def test_s3_bucket_public_read_disabled(self, sh):
+        """Tests if S3 Bucket public read has been turned off
+        
+        Arguments:
+            s3_test_bucket_public_read {string} -- S3 bucket name
+            sh {SecurityHubRules} -- Instance of class SecurityHubRules
+        """
+
+        # create bucket
+        sh.client_s3.create_bucket(ACL="public-read", Bucket="test")
+
+        # test s3_bucket_public_read_prohibited function
+        sh.s3_bucket_public_read_prohibited("test")
+
+        # validate test
+        response = sh.client_s3.get_bucket_acl(Bucket="test")
+        assert response["Grants"][0]["Permission"] == "FULL_CONTROL"
+
+    def test_invalid_bucket(self, sh):
+        # create bucket
+        sh.client_s3.create_bucket(ACL="public-read", Bucket="test")
+
+        # validate test
+        assert not sh.s3_bucket_public_read_prohibited("test123")
```

Comment:
Comment only: 
I think this gives a clue for how we should refactor the error handling - it should raise the NoSuchBucket exception (or whatever) directly and we should assert that that exception is raised and caught here.

That means we should have the try except in the lambda_handler remediate() function, I think

Reply:
Doesn't that mean we'll be catching 20 different exceptions in the `lambda_handler`?